execute/plugin.go

```diff
@@ -243,8 +244,12 @@ func (p *Plugin) Observation(
 			groupedCommits[report.SourceChain] = append(groupedCommits[report.SourceChain], report)
 		}
 
-		// TODO: Fire off messages for an attestation check service.
-		return exectypes.NewObservation(groupedCommits, messages, nil).Encode()
+		tkData, err1 := p.tokenDataObserver.Observe(ctx, messages)
```

Comment:
Is it going to be synchronous?

e.g. If Circle's API is slow, it affects our observation duration?

Reply:
It's gonna be implementation details, for the plugin perspective we behave like it's synchronous, plugin shouldn't know that. Under the hood we will figure out background processing etc